case-lib/pipeline.sh

```diff
@@ -2,6 +2,19 @@
 
 func_pipeline_export()
 {
+    # For legacy HDA platform, there is no topology, we have to exporting pipeline
+    # information from proc file system.
+    is_sof_used || {
```

Comment:
will this test work where we have dmi info available and the card name doesnt contain "sof"? like the chromebooks?

Reply:
For example : on my helios, card name is GoogleHeliosRev1. There's no sof prefix there.

Reply:
On the other hand, the procfs info for PCMs is valid for all cases. Why not just use that instead of topology?

Reply:
> On the other hand, the procfs info for PCMs is valid for all cases. Why not just use that instead of topology?

I guess part of the answer is in this code comment:

```
for pcm in card_info['pcm']:
                # There are limited pipeline parameters in the proc file system,
                # add some default parameters to make use of sof-test for legacy HDA test
                pcm['fmt'] = 'S16_LE'
                pcm['fmts'] = 'S16_LE S24_LE S32_LE'
                ...
```

Reply:
@ranj063 I guess you are talking about Chromebook under ChromeOS,  because the content of /proc/asound/cards on my Helios looks like this:
```
ubuntu@helios:~$ cat /proc/asound/cards
 0 [sofcmlrt1011rt5]: sof-cml_rt1011_ - sof-cml_rt1011_rt5682
                      Google-Helios-rev2
```
Currently, sof-test cannot run on ChromeOS, because most infrastructure level python scripts cannot run on that OS.

Reply:
> @ranj063 I guess you are talking about Chromebook under ChromeOS, because the content of /proc/asound/cards on my Helios looks like this:
> 
> ```
> ubuntu@helios:~$ cat /proc/asound/cards
>  0 [sofcmlrt1011rt5]: sof-cml_rt1011_ - sof-cml_rt1011_rt5682
>                       Google-Helios-rev2
> ```
> 
> Currently, sof-test cannot run on ChromeOS, because most infrastructure level python scripts cannot run on that OS.

@aiChaoSONG no Im talking about linux with nocodec mode

Reply:
I dont know why but yesterday my card name was GoogleHeliosrev2 but its back to sof-nocodec now

Reply:
> Currently, sof-test cannot run on ChromeOS, because most infrastructure level python scripts cannot run on that OS.

In theory the `dev_install` command can install python
https://chromium.googlesource.com/chromiumos/platform2/+/refs/heads/master/dev-install/

It does NOT mean everything would work but it would be a good start.

Reply:
> @ranj063 I guess you are talking about Chromebook under ChromeOS, because the content of /proc/asound/cards on my Helios looks like this:
> 
> ```
> ubuntu@helios:~$ cat /proc/asound/cards
>  0 [sofcmlrt1011rt5]: sof-cml_rt1011_ - sof-cml_rt1011_rt5682
>                       Google-Helios-rev2
> ```
> 
> Currently, sof-test cannot run on ChromeOS, because most infrastructure level python scripts cannot run on that OS.

that's not a good-enough reason to screw-up the card name parsing. 

I think you guys are confusing things, the syntax is

cardIndex [mixer_name]: driver_name - card_name
          long_card_name

@ranj063 can you provide what you have on your device w/ Chrome?

Reply:
@plbossart this is what we see on chrome: 
localhost ~ # cat /proc/asound/cards
 0 [sofrt5682      ]: sof-rt5682 - sof-rt5682
                      Google-Delbin-1.0

Reply:
but for some reason when I was testing yesterday my card name was as below on my helios. I am unable to reproduce this. So I do not know where this was coming from. 
root@ranjani-Helios:~# cat /proc/asound/cards
 0 [GoogleHeliosrev]:  -
                      Google-Helios-rev2